litestar/handlers/base.py

```diff
@@ -203,6 +203,14 @@ def _get_merge_opts(self, others: tuple[Router, ...]) -> dict[str, Any]:
                 merge_opts.get("signature_namespace", {}), other.signature_namespace
             )
 
+            # TODO(tofran): properly implement layered security
+            current_security = merge_opts.get("security")
+            if other.security is not None:
+                if current_security is None:
+                    merge_opts["security"] = other.security
+                else:
+                    merge_opts["security"] = (*current_security, *other.security)
```

Comment:
This can be simplified like the other merge opts, e.g. `middleware`:
```suggestion
            merge_opts["security"] = (*other.security, *merge_opts.get("security", ()))
```